frontend/src/metabase/dashboard/hoc/controls/hooks/use-refresh-dashboard.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  fetchDashboard,
+  fetchDashboardCardData,
+} from "metabase/dashboard/actions";
+import { useDispatch } from "metabase/lib/redux";
+import type { DashboardId } from "metabase-types/api";
+
+export const useRefreshDashboard = ({
+  dashboardId,
+  queryParams,
+}: {
+  dashboardId: DashboardId;
+  queryParams: Record<string, any>;
+}): {
+  refreshDashboard: () => Promise<void>;
+} => {
+  const dispatch = useDispatch();
+
+  const refreshDashboard = async () => {
+    if (dashboardId) {
+      await dispatch(
+        fetchDashboard({
+          dashId: dashboardId,
+          queryParams: queryParams,
+          options: { preserveParameters: true },
+        }),
+      );
+      dispatch(
+        fetchDashboardCardData({
+          isRefreshing: true,
+          reload: true,
+          clearCache: false,
+        }),
+      );
```

Comment:
`fetchDashboardCardData` relies on the dashboard being loaded, so they have to run in parallel. It's a little annoying since we can't see the returned value but it's how it is